runtimes/spiritnet/src/lib.rs

```diff
@@ -623,13 +756,43 @@ construct_runtime! {
 
 		// KILT Pallets. Start indices 60 to leave room
 		KiltLaunch: kilt_launch::{Pallet, Call, Storage, Event<T>, Config<T>} = 60,
+		Ctype: ctype::{Pallet, Call, Storage, Event<T>} = 61,
+		Attestation: attestation::{Pallet, Call, Storage, Event<T>} = 62,
+		Delegation: delegation::{Pallet, Call, Storage, Event<T>} = 63,
+		Did: did::{Pallet, Call, Storage, Event<T>, Origin<T>} = 64,
 
 		// Parachains pallets. Start indices at 80 to leave room.
 		ParachainSystem: cumulus_pallet_parachain_system::{Pallet, Call, Storage, Inherent, Event<T>, Config} = 80,
 		ParachainInfo: parachain_info::{Pallet, Storage, Config} = 81,
 	}
 }
 
+impl did::DeriveDidCallAuthorizationVerificationKeyRelationship for Call {
+	fn derive_verification_key_relationship(&self) -> Option<did::DidVerificationKeyRelationship> {
+		match self {
+			Call::Attestation(_) => Some(did::DidVerificationKeyRelationship::AssertionMethod),
+			Call::Ctype(_) => Some(did::DidVerificationKeyRelationship::AssertionMethod),
+			Call::Delegation(_) => Some(did::DidVerificationKeyRelationship::CapabilityDelegation),
+			// DID creation is not allowed through the DID proxy.
+			Call::Did(did::Call::create(..)) => None,
+			Call::Did(_) => Some(did::DidVerificationKeyRelationship::Authentication),
+			//TODO: add a batch call case that returns the right key type if all calls in the batch require the same
```

Comment:
Q: Can we handle this later or should this be fixed before adding KILT features to Spiritnet?